prime-router/docs/design/design/translations.md

```diff
@@ -0,0 +1,77 @@
+
+### FHIR-to-FHIR configurations
+
+FHIR to FHIR transform templates are configured as `.yml` files which are presently located in the repository under `prime-router/metadata/fhir_transforms`. Whereas HL7-to-FHIR and FHIR-to-HL7 templates use a library on top of the respective HAPI library to perform the translations, FHIR to FHIR transforms are performed using the HAPI FHIR library directly, with the logic for loading templates contained in `ConfigSchemaReader.kt` and the logic for using the loaded templates to perform the translation contained in `FhirTransformer.kt`.
+
+#### Template Structure
+
+For those familiar with the templates of [LinuxForHealth HL7 to FHIR Converter library](https://github.com/LinuxForHealth/hl7v2-fhir-converter#linuxforhealth-hl7-to-fhir-converter), the FHIR-to-FHIR templates will look like a simpler version. This is because, at their core, they are simply selecting a particular resource in a bundle and setting it to a particular value.
+
+##### Template Elements
+
+An "element" is the basic building block of a template and contains a set of parameters that describe what bundle property to set, under what conditions to set it to, and what value to set it to. The `FhirTransformer` class traverses each element in the template and applies the transform following the rules specified by a particular element. The basic fields that make up an element are as follows:
+
+- **condition**: an expression evaluating to true or false. When true, the transform will be applied
+- **resource**: the FHIR path of the resource in the bundle that is to have its property/properties modified
+- **bundleProperty**: The FHIR path to the property of the resource returned by the resource parameter above
+- **value**: The value to set the bundleProperty to
+
+Example Element:
+```yml
+  - name: processing-id
+    condition: 'Bundle.entry.resource.ofType(MessageHeader).meta.tag.code.empty()'
+    resource: 'Bundle.entry.resource.ofType(MessageHeader)'
+    bundleProperty: '%resource.meta.tag.code'
+    value: [ '"P"' ]
+```
+
+As seen in the example above, each element can have a name, which is used purely for the purposes of extending or overriding values from [extended templates](#extending-templates).
+
+##### Template Constants
+
+Each template can define a set of constants to use throughout its elements in order to make the template easier to read. Constants are defined by populating the `constants` property. Like so:
+
+```yml
+constants:
+  rsext: '"https://reportstream.cdc.gov/fhir/StructureDefinition/"'
+```
+
+The constants can be used in any FHIR expression belonging to any element in the template or any child template. When a constant appears in a special string enclosed in %\`\`, the code will look for strings defined as constants and replace them with their values. Example: %\`rsext-software-vendor-org\`
+
+##### Extending Templates
```

Comment:
I thought I left a comment here, but I must have not hit save. Basically just wanted to say we can add a lot more detail here mostly around:

- what does it mean to extend a schema?
- what does it mean to reference a schema in an element? what does this look like?
- how are these two things different and when should you do one or another?

I've gone through it before, and it's honestly pretty confusing, so it would be good to have some kind of reference for that stuff. That said, this is a good starting point, and we can fill in more details over time or as part of another ticket.